src/WebJobs.Script/Binding/HttpBinding.cs

```diff
@@ -98,16 +106,24 @@ public override Collection<CustomAttributeBuilder> GetCustomAttributes(Type para
                     Content = new StringContent((string)content)
                 };
             }
+            else if (content is byte[])
+            {
+                // same for byte arrays
+                response = new HttpResponseMessage(statusCode)
+                {
+                    Content = new ByteArrayContent((byte[])content)
```

Comment:
Why are we special casing `byte[]` here and not letting it be handled below? Have you tested this code without setting explicit headers from your function to inspect the behavior?